mcp: changed streamID's from int64 to random strings

2bitburrito opened the pull request:
- Changed StreamID's to store randomly generated strings. 
- Created a defaultStreamID field on the StreamableServerTransport struct which is generated and saved on instansiation.
- Updated all tests.

Fixes #259

jba commented on mcp/streamable.go:
```diff
@@ -199,12 +200,11 @@ func (t *StreamableServerTransport) SessionID() string {
-	incoming  chan jsonrpc.Message // messages from the client to the server
-	done      chan struct{}
+	sessionID       string
+	defaultStreamID StreamID
```
I think we'd just use the empty string for this.

jba commented on mcp/streamable.go:
```diff
@@ -283,9 +283,10 @@ func signalChanPtr() *chan struct{} {
 
-// A StreamID identifies a stream of SSE events. It is unique within the stream's
+// A StreamID identifies a stream of SSE events. It is a random string
+// 26 characters in length
```
Remove the sentence about the random string—that's an implementation detail. Add a sentence saying that it's globally unique.

jba commented on mcp/streamable_test.go:
```diff
@@ -222,9 +222,10 @@ func TestServerInitiatedSSE(t *testing.T) {
-	client := NewClient(testImpl, &ClientOptions{ToolListChangedHandler: func(ctx context.Context, cc *ClientSession, params *ToolListChangedParams) {
-		notifications <- "toolListChanged"
-	},
+	client := NewClient(testImpl, &ClientOptions{
```
This change seems unrelated, and it's a no-op; revert it.

2bitburrito commented:
Ok - thanks for the notes. I apprecaite the feedback and apologies if there are any further issues. I haven't done much open-source stuff before. 

I have: 
- Removed the defaultStreamID implementation and just used an empty string for the default stream.
- Updated the comment to not include implementation details
- Fixed the auto-formatted issue
- Added an extra test in TestEventID that ensure's StreamID's can be empty strings.

Thanks, let me know how that looks and whether there's anything else you'd recommend. 

Hugh

jba commented on mcp/streamable.go:
```diff
@@ -182,7 +182,7 @@ func NewStreamableServerTransport(sessionID string, opts *StreamableServerTransp
 		requestStreams: make(map[jsonrpc.ID]StreamID),
 	}
-	t.streams[0] = newStream(0)
+	t.streams[StreamID("")] = newStream(StreamID(""))
```
You shouldn't need to write StreamID in either place.

jba commented on mcp/streamable.go:
```diff
@@ -242,7 +240,7 @@ type StreamableServerTransport struct {
 type stream struct {
 	// id is the logical ID for the stream, unique within a session.
-	// ID 0 is used for messages that don't correlate with an incoming request.
+	// defaultStreamID is used for messages that don't correlate with an incoming request.
```
The empty string is used...

2bitburrito commented:
Hi Jonathan, I've made those changes now.
